.dockerignore

```diff
@@ -0,0 +1,29 @@
+# See https://docs.docker.com/engine/reference/builder/#dockerignore-file for more about ignoring files.
+
```

Comment:
This file is a translation of `.gitignore`, plus it ignores the `.git/` folder to avoid needlessly copying all git branches and configuration into the build image. The one piece of Git metadata that is needed, the revision ref, is passed as a build arg/environment variable. I believe this to be a pretty well-established approach to using git metadata in Docker image build steps.